File: java/src/main/java/problems301_400/RansomNote.java
package problems301_400;

enum RansomNote {
    DEFAULT {
        @Override
        boolean canConstruct(String ransomNote, String magazine) {
            int[] map = new int[128];
            for (char c : ransomNote.toCharArray()) {
                map[c]++;
            }
            for (char c : magazine.toCharArray()) {
                map[c]--;
            }
            for (int i : map) {
                if (i > 0) {
                    return false;
                }
            }
            return true;
        }
    };

    abstract boolean canConstruct(String ransomNote, String magazine);
}

File: kotlin/src/main/kotlin/problems301_400/RansomNote.kt
package problems301_400

enum class RansomNote {
    DEFAULT {
        override fun canConstruct(ransomNote: String, magazine: String): Boolean {
            val map = IntArray(128)
            for (c in ransomNote.toCharArray()) {
                map[c.code]++
            }
            for (c in magazine.toCharArray()) {
                map[c.code]--
            }
            for (i in map) {
                if (i > 0) {
                    return false
                }
            }
            return true
        }
    },
    ;

    abstract fun canConstruct(ransomNote: String, magazine: String): Boolean
}
